Issue #7 - Refactoring of code smells identified by sonarqube for outdoor directions

NicholasNagy opened the pull request:
In this pull request, I refactored many of the code smells related to files in the outdoor directions feature. The biggest refactorings were related to the cognitive complexities of methods.

NicholasNagy committed:
Refactor parseDirections

I refactored a very large method called parseDirections as it was very large and had a very big cognitive complexity (37).

NicholasNagy committed:
Merge branch 'chain_directions' of https://github.com/RobertBeaudenon/SOEN390-CampusGuideMap into chain_directions

NicholasNagy committed:
Refactoring out code smells in map fragment

I refactored out the following code smells:
- Added nested comments to unimplemented overridden function to explain why they are empty
- Refactored out setting the button functionality in the showInstructions method because the cognitive complexity was too large (18 instead of 15)
- Merged the if statements into line 388 since there was no use in having them nested

NicholasNagy committed:
Refactor fetchRouteTimes

Refactored method with high cognitive complexity (26) by handling the route response in a separate method.

NicholasNagy committed:
Refactored code smell induced by previous refactor

My previous refactor to lower the code smell of a method in fact had induced a new smell due to high cognitive complexity of the new method, so I more logically broke it down into 2 new methods for each button that we want to set the functionality for.

NicholasNagy committed:
Refactoring Map Repository

I refactored the map repository branches to be broken down into the sgw and loyola buildings

In order to reduce the cognitive complexity of the getBuildingsFromJSON, I refactored out getting the json array since, the functionality was not super related to the rest of the method.

NicholasNagy committed:
Merge remote-tracking branch 'origin/master' into outdoorDirections

NicholasNagy committed:
Removing TODO's

I don't know why it mentions to bring the camera to the current location when showing a path (I'm guessing that the one who put it there meant the start location), but we decided that if the user decides to leave the fragment and start looking at other things, that we wouldn't force the user to follow the path.

The next todo mentions adding events to the buildings' information,  but we don't even consider that in our road map, so I will remove it.

makchamp commented on app/src/main/java/com/droidhats/campuscompass/repositories/NavigationRepository.kt:
```diff
@@ -291,62 +304,136 @@ class NavigationRepository(private val application: Application) {
+     * @param mode of transportation
+     * @param waypoints
+     * @param instructions
+     * @param intCoordinates
```
should reorder param comments to be same as the passed params

makchamp commented on app/src/main/java/com/droidhats/campuscompass/views/MapFragment.kt:
```diff
@@ -740,12 +751,13 @@ class MapFragment : Fragment(), OnMapReadyCallback, GoogleMap.OnMarkerClickListe
 
-    override fun onCalendarEventClick(item: CalendarEvent?) {}
+    override fun onCalendarEventClick(item: CalendarEvent?) {
+        // add no functionality for this overidden function
```
typo in word overriden

makchamp commented on app/src/main/java/com/droidhats/campuscompass/repositories/NavigationRepository.kt:
```diff
@@ -291,62 +304,136 @@ class NavigationRepository(private val application: Application) {
+     * @param intCoordinates list of coordinates along the path
+     * @param mode of transportation
+     */
+    fun addMainInstructions(
```
function could be private

makchamp commented on app/src/main/java/com/droidhats/campuscompass/repositories/NavigationRepository.kt:
```diff
@@ -291,62 +304,136 @@ class NavigationRepository(private val application: Application) {
+     * @param intCoordinates list of coordinates along the path
+     * @param mode of transportation
+     */
+    fun parseIntoArray(
```
function could be private

makchamp commented on app/src/main/java/com/droidhats/campuscompass/repositories/NavigationRepository.kt:
```diff
@@ -291,62 +304,136 @@ class NavigationRepository(private val application: Application) {
+     * @param intCoordinates
+     * @param stepsArray
+     */
+    fun addEndInstruction(
```
function could be private

makchamp reviewed (approved):
Looks good to go, just a few small things to tweak

NicholasNagy committed:
Fixing changes requested in review

I changed some encapsulation requested during the pull request review, and a typo

makchamp commented:
looks good baby
